apps/web/src/pages/CreateApplicationPage/CreateApplicationPage.tsx

```diff
@@ -85,10 +85,10 @@ const CreateApplication = () => {
     path: string,
     toastFunction: () => void,
   ): Promise<void> => {
-    if (navigateWithToastCounter.current > 0) return; // we've already started navigation
+    navigateWithToastCounter.current += 1;
+    if (navigateWithToastCounter.current > 1) return; // we've already started navigation
     await navigate(path, { replace: true });
     toastFunction();
-    navigateWithToastCounter.current += 1;
   };
 
```

Comment:
Not entirely sure why or when this started, but it appeared this function in question kept firing many, many times. 
The counter incrementing after the async function might've caused the function to fire many times before the current actually incremented. Moving the counter up ensures it readily and immediately updates and doesn't get held up by the other functions.

Reply:
Yeah, good find.  You can find the diff in the "Upgrade to React Router 7" PR.  As you said, it's that an `await` was added to the navigation so the execution never got to the counter increment.
![image](https://github.com/user-attachments/assets/cf1615e5-f8c3-4804-877a-342607a56587)